bpf-gpl/nat.h

```diff
@@ -444,9 +448,14 @@ static CALI_BPF_INLINE int is_vxlan_tunnel(struct iphdr *ip)
 
 static CALI_BPF_INLINE bool vxlan_v4_encap_too_big(struct __sk_buff *skb)
 {
-	__u32 mtu = TNNL_INNER_ETH_MTU;
+	__u32 mtu = TUNNEL_MTU;
 
-	if (skb->len > mtu) {
+	/* RFC-1191: MTU is the size in octets of the largest datagram that
+	 * could be forwarded, along the path of the original datagram, without
+	 * being fragmented at this router.  The size includes the IP header and
+	 * IP data, and does not include any lower-level headers.
+	 */
+	if (skb->len - sizeof(struct ethhdr) > mtu) {
```

Comment:
Paranoia: can an SKB have len less than sizeof(struct ethhdr)?  If so it would be safer to write `skb->len > mtu + sizeof(struct ethhdr)` to avoid a possible underflow.

Reply:
ack